stacks-signer/src/client/stacks_client.rs

```diff
@@ -180,6 +180,29 @@ impl StacksClient {
         Ok(signer_slots)
     }
 
+    /// Get the stackerdb signer slots for a specific reward cycle
+    pub fn get_parsed_signer_slots(
+        &self,
+        reward_cycle: u64,
+    ) -> Result<HashMap<StacksAddress, SignerSlotID>, ClientError> {
+        let signer_set =
+            u32::try_from(reward_cycle % 2).expect("FATAL: reward_cycle % 2 exceeds u32::MAX");
+        let signer_stackerdb_contract_id = boot_code_id(SIGNERS_NAME, self.mainnet);
+        // Get the signer writers from the stacker-db to find the signer slot id
+        let stackerdb_signer_slots =
+            self.get_stackerdb_signer_slots(&signer_stackerdb_contract_id, signer_set)?;
+        let mut signer_slot_ids = HashMap::with_capacity(stackerdb_signer_slots.len());
+        for (index, (address, _)) in stackerdb_signer_slots.into_iter().enumerate() {
+            signer_slot_ids.insert(
+                address,
+                SignerSlotID(
+                    u32::try_from(index).expect("FATAL: number of signers exceeds u32::MAX"),
+                ),
+            );
+        }
```

Comment:
Maybe nicer to `.collect()` here? Up to you

```suggestion
        let signer_slot_ids = stackerdb_signer_slots
            .into_iter()
            .enumerate()
            .map(|(index, (address, _))| (
                address,
                SignerSlotID(
                    u32::try_from(index).expect("FATAL: number of signers exceeds u32::MAX"),
                ),
            ))
            .collect();
```